### PR TITLE
Add `postInstall()` method.

### PR DESCRIPTION
**Pull request by rob006**
| Q | A |
| --- | --- |
| Is bugfix? | no |
| New feature? | yes |
| Breaks BC? | no |
| Fixed issues | - |

This allows to define `post-install-cmd` scripts in the same manner as for `post-create-project-cmd`.

### CODE REVIEW

**Comment by rob006** [comment]
Rebased.

**Comment by samdark** [comment]
What are use cases for these?

**Comment by rob006** [comment]
I hope to use it in https://github.com/yiisoft/yii2-app-basic/pull/60 to run some commands on app install. Assuming that we don't store whole config in VCS, we may want run some commands after composer install instead of create-project (`post-create-project-cmd` will not run if we just checkout existing project).

File: Installer.php [hunk 1]
```diff
@@ -228,12 +228,22 @@ protected function removeBaseYiiFiles()
             rmdir($yiiDir);
         }
     }
-    
+
```
Comment (dynasource) [inline Installer.php]: Dont we want method descriptions here?

File: Installer.php [hunk 2]
```diff
@@ -228,12 +228,22 @@ protected function removeBaseYiiFiles()
...
+    {
+        static::runCommands($event, __METHOD__);
+    }
+
     public static function postCreateProject($event)
     {
+        static::runCommands($event, __METHOD__);
+    }
+
+    protected static function runCommands($event, $extraKey)
```
Comment (dynasource) [inline Installer.php]: and here. What kind of type is $event?

**Comment by rob006** [comment]
Done.

**Comment by cebe** [comment]
Thank you!

**Pull request merged**